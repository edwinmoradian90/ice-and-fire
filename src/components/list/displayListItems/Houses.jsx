import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { convertName } from '../../../utils/helpers'
import axios from 'axios'
import { Container, Text } from '../../../utils/styledComponents/main'

export default function Houses(props) {
    const [heirName, setHeirName] = useState('')
    const [heirId, setHeirId] = useState('')
    const [currentLordId, setCurrentLordId] = useState('')
    const [currentLordName, setCurrentLordName] = useState('')
    const [founderName, setFounderName] = useState('')
    const [founderId, setFounderId] = useState('')

    const {
        name,
        region,
        heir,
        founded,
        founder,
        currentLord,
    } = props.items.data || 'loading...'

    const splitUrl = url => {
        if (url) {
            const newUrl = url.split('/')
            return newUrl[newUrl.length - 1]
        }
    }

    const isEmpty = item => {
        if (item === '' || item === null) {
            return 'N/A'
        }
        return item
    }

    useEffect(() => {
        if (heir) {
            axios.get(heir)
                .then(res => {
                    const { name } = res.data
                    setHeirName(name)
                    setHeirId(splitUrl(heir))
                })
                .catch(err => console.log(err))
        }
        if (currentLord) {
            axios.get(currentLord)
                .then(res => {
                    const { name } = res.data
                    setCurrentLordName(name)
                    setCurrentLordId(splitUrl(currentLord))
                })
                .catch(err => console.log(err))
        }

        if (founder) {
            axios.get(founder)
                .then(res => {
                    const { name } = res.data
                    setFounderName(name)
                    setFounderId(splitUrl(founder))
                })
                .catch(err => console.log(err))
        }

    }, [currentLord, heir, founder])

    return (
        <Container
            className="houses"
            height="100vh"
            justifyContent="flex-start"
            alignItems="flex-start"
        >
            <Text className="name">House name: {isEmpty(name)}</Text>
            <Text className="region">House region: {isEmpty(region)}</Text>
            <Text className="founder">
                {founderName ? (
                    <>
                        House Founder: {' '}
                        <Link to={`/characters/${founderId}/${convertName(founderName)}`}>
                            {founderName}
                        </Link>
                    </>
                ) : (
                        <p className="founderName">House Founder: {isEmpty(founderName)}</p>
                    )}
            </Text>
            <Text className="founded">Founded During: {isEmpty(founded)}</Text>
            {heirName ? (
                <Text className="heir">
                    House heir: {' '}
                    <Link
                        className='heirName'
                        to={`/characters/${heirId}/${convertName(heirName)}`}
                    >
                        {heirName}
                    </Link>
                </Text>
            ) : (
                    <Text className="heirName"> House heir: N/A</Text>
                )
            }
            <Text className="currentLord">
                Current Lord:
                {currentLord
                    ?
                    (
                        <Link
                            className='currentLordName'
                            to={`/characters/${currentLordId}/${convertName(currentLordName)}`}
                        >
                            {currentLordName || 'loading...'}
                        </Link>
                    )
                    :
                    (
                        <span className="currentLordName">{isEmpty(currentLordName)}</span>
                    )
                }
            </Text>
        </Container >
    )
}

const { string } = PropTypes
Houses.propTypes = {
    name: string,
    founded: string,
    founder: string,
    currentLord: string,
    heir: string,
    region: string,
}