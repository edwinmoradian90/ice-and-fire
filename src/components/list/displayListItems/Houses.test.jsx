import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Houses from './Houses';

describe('Houses', () => {
    const items = {
        data: {
            name: 'someone',
            region: 'somewhere',
            heir: 'someone',
            founded: 'sometime',
            founder: 'someone',
            currentLord: 'someone',
        },
    };
    it('renders correctly', () => {
        const component = renderer.create(
            <Router>
                <Houses items={items} />
            </Router>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});