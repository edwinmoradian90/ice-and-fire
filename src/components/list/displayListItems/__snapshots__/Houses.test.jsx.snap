// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Houses renders correctly 1`] = `
<div
  className="sc-AxirZ lmlOFf houses"
  height="100vh"
>
  <div
    className="sc-AxhUy fClwJB name"
  >
    House name: 
    someone
  </div>
  <div
    className="sc-AxhUy fClwJB region"
  >
    House region: 
    somewhere
  </div>
  <div
    className="sc-AxhUy fClwJB founder"
  >
    <p
      className="founderName"
    >
      House Founder: 
      N/A
    </p>
  </div>
  <div
    className="sc-AxhUy fClwJB founded"
  >
    Founded During: 
    sometime
  </div>
  <div
    className="sc-AxhUy fClwJB heirName"
  >
     House heir: N/A
  </div>
  <div
    className="sc-AxhUy fClwJB currentLord"
  >
    Current Lord:
    <a
      className="currentLordName"
      href="/characters//"
      onClick={[Function]}
    >
      loading...
    </a>
  </div>
</div>
`;
